### ISSUE TITLE
[BUG] Standard users are unable to force delete a node from a downstream clusters that they provisioned

### ISSUE THREAD

**Issue by markusewalker** [rancher/rancher#41698]
**Rancher Server Setup**
- Rancher version: `v2.7-head` https://github.com/rancher/rancher/commit/5334be362e291576c8f32b4432762399729832e6
- Installation option (Docker install/Helm Chart): Docker
   - If Helm Chart, Kubernetes Cluster and version (RKE1, RKE2, k3s, EKS, etc):
- Proxy/Cert Details:

**Information about the Cluster**
- Kubernetes version: `v1.26.4+rke2r1`
- Cluster Type (Local/Downstream): Downstream
   - If downstream, what type of cluster? (Custom/Imported or specify provider for Hosted/Infrastructure Provider): Linode

**User Information**
- What is the role of the user logged in? (Admin/Cluster Owner/Cluster Member/Project Owner/Project Member/Custom) Cluster Owner
  - If custom, define the set of permissions:

**Describe the bug**

While validating issue https://github.com/rancher/rancher/commit/5334be362e291576c8f32b4432762399729832e6, I noted that when you attempt to force delete a node as a standard user, you receive the following error:

![image](https://github.com/rancher/rancher/assets/52591819/fc44f6fb-67ca-40d6-b8ff-93aa0272c7bd)

This occurs because the default standard user that Rancher creates does not have access to make changes to the backend infrastructure. As such, you will need to have the correct RBAC permissions assigned or switch to an administrative user. As a standard user provisioning a cluster, I would expect to be able to make these operations on my downstream cluster without relying on an administrator.

**To Reproduce**

1. As a standard user, provision a downstream RKE2 node driver cluster (1 etcd, 1 cp, 1 worker).
     - i used `v1.26.4+rke2r1`, but the K8s version should not matter.
2. On the node provider's infrastructure side, delete the cloud credential created to provision the cluster.
3. Back in Rancher > Cluster Management > RKE2 cluster, delete the cluster; validate that it is stuck in `Deleting` state.
4. Click the RKE2 cluster > ellipses and click `Force Delete` on the worker node. Note that you see the above-mentioned error message as a standard user.
5. Log out and log back in as an admin user.
6. Attempt to force delete the worker node and validate that you are able to.

**Result**
Standard users that are cluster owners are unable to delete downstream clusters.

**Expected Result**

Standard users that are cluster owners should be able to delete downstream clusters.

**Comment by thatmidwesterncoder** [comment]
Upon more investigation - this issue is going to be a bit bigger than it looked. 

This appeared to be maybe an issue with just the `cluster-owner` role initially but when investigating why the `rke-machine.cattle.io` wild card role(s) for the cluster either wasn't getting picked up by the backend or whether it was the wrong permission it seems that the main issue is something along the lines of how we're handling permissions on the `cluster.management.cattle.io` resources in general.

tl;dr: `fleet-default` contains all of the cluster resources for now, we need them to be in separate namespaces in order to be able to assign the "right" role bindings and give permissions to edit the cluster directly.

---

It appears when clicking `force-delete` the UI does a few operations, namely a `list` on the rke-machines.cattle.io resources and then a `get` to get the one needed. These are probably well and good since they're read-only - but the problem then becomes to force-delete it it must either apply an annotation or something (assumed), which needs direct edit access via a PUT request. 

This gets into weird territory since technically speaking by adding `update` to the permissions that would allow _any_ user to edit _any_ cluster in rancher since they all live in fleet-default. After bringing this up in standup/afterparty the best way forward would require a couple things:

1. Clusters and therefore rke-machines would need to be moved to their own namespaces rather than fleet-default
2. There should probably be a controller that does all of the rbac-related things to create all of the role binding resources, users, etc. Right now thats all handled in other controllers afaik. 

Moving the cluster resources + role bindings to separate namespaces will be one thing; I think it's in the works - but after that is done the refactor of pulling out the role management into its own controller will be a rather large change. But it would end up being significantly better than the current setup where I assume the rolebindings are just created directly when managing user roles.
